### ISSUE TITLE
MS Word: Formatting toggle incorrectly announced when cursor in front of text

### ISSUE THREAD

**Issue by michaelweghorn** [nvaccess/nvda#16412]
### Steps to reproduce:

1. start NVDA
2. start MS Word
3. type 
4. Type "hello world"
5. move text cursor to the beginning of the text
6. press Ctrl+B (in English UI) to enable bold formatting (for newly typed text)

### Actual behavior:

NVDA announces "Bold off", even though bold is now enabled for newly typed text, as the button in the toolbar/ribbon shows.

Interestingly, the Speech Viewer output seems to first have the correct, then the incorrect state (and maybe only the last one is announced?):

```
ctrl+b
Bold on
Bold Off
```

### Expected behavior:

NVDA should announce "Bold on".

#### NVDA installed/portable/running from source:

running from source

#### NVDA version:

git master as of commit 11a2b4f1dba2d1503c61a32a41124929fb7c4702

#### Windows version:

Windows 10 22H2 (OS Build 19045.4291)

#### Name and version of other software in use when reproducing the issue:

Microsoft® Word LTSC MSO (16.0.14332.20678) 64-bit 
(Microsoft Office LTSC Professional Plus 2021)

#### Other information about your system:

n/a

#### Does the issue still occur after restarting your computer?

yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

no

#### If NVDA add-ons are disabled, is your problem still occurring?

n/a

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

n/a

**Comment by CyrilleB79** [comment]
Not tested but the issue is very likely an erroneous UIA notification from Word.

What is the result with Narrator? If "bold off" is heard, the issue should be reported to Microsoft.

Regarding the impact of this issue in NVDA. Have you identified other cases where this notification is incorrect? In the past, it was incorrect when between two spaces, but it has been fixed since then.
What is the most general case in which this issue occurs according to you?

**Comment by CyrilleB79** [comment]
OK. If no one can test, I'll test on a more recent Office version tonight to see if it is fixed.

**Comment by CyrilleB79** [comment]
Actually your build number is quite old, probably because of LTSC channel.

On my side, the issue seems fixed with Microsoft® Word 2016 MSO (Version 2403 Build 16.0.17425.20176) 32 bits.

Thus I'll close this issue since the fix has been released on MS side.

**Comment by michaelweghorn** [comment]
Thanks for checking! Good to hear that this is fixed in newer builds.
